### PR TITLE
OData error with target and details #76

### PR DESCRIPTION
**Pull request by congysu**
Add target and details to OData error payload.
Add writer and reader tests.
Update public API baseline.

### CODE REVIEW

**Comment by congysu** [comment]
@OData/odata-team Please help to review this PR.  Thanks.

File: src/Microsoft.OData.Core/ODataError.cs [hunk 1]
```diff
@@ -39,6 +39,17 @@ public string Message
             set;
         }
 
+        /// <summary>Gets or sets the target of the particular error.</summary>
+        /// <returns>For example, the name of the property in error</returns>
+        public string Target { get; set; }
+
+        /// <summary>
+        /// An array of JSON objects that MUST contain name/value pairs for code and message, and MAY contain
```
Comment (lewischeng-ms) [inline src/Microsoft.OData.Core/ODataError.cs]: A collection of JSON objects?

**Comment by lewischeng-ms** [comment]
Looks good to me.

File: src/Microsoft.OData.Core/ODataError.cs (hunk 1 above)
Comment (congysu) [inline src/Microsoft.OData.Core/ODataError.cs]: Done.

**Comment by congysu** [comment]
Checked-in 103b6a4c30316ada5f6fab1c0b2b33fe0491339d - 20151c21fe1fd176aaf5ea62f6598e794f30cc61.

**Pull request closed without merging**